src/baseline/ast/expr.rs

```diff
@@ -1443,10 +1467,12 @@ where
             Intrinsic::IntMul => self.asm.int_mul(MachineMode::Int32, dest, lhs, rhs),
             Intrinsic::IntDiv => self.asm.int_div(MachineMode::Int32, dest, lhs, rhs),
             Intrinsic::IntMod => self.asm.int_mod(MachineMode::Int32, dest, lhs, rhs),
+            Intrinsic::IntNeg => self.asm.int_neg(MachineMode::Int32, dest, lhs),
```

Comment:
Can you please move these unary intrinsics out of `emit_intrinsic_bin`? `lhs`, `rhs` and `BinOp` doesn't make sense for them.

Reply:
Agreed, will do. :-)

Reply:
Also did the same for BoolNot.